### PR TITLE
Fix rules engine trying to load tasks for undefined contact.

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

medic/cht-core#7041

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
hey @latin-panda . Mind having a look? Thanks!

File: webapp/src/ts/services/contact-view-model-generator.service.ts [hunk 1]
```diff
@@ -164,7 +164,8 @@ export class ContactViewModelGeneratorService {
       .then(doc => {
         children.push({
           doc: doc,
-          isPrimaryContact: true
+          isPrimaryContact: true,
+          id: doc._id,
```
Comment (latin-panda) [inline webapp/src/ts/services/contact-view-model-generator.service.ts:168]:
```suggestion
          id: doc?._id,
```
not sure if doc can be undefined here

**Review by latin-panda — APPROVED** [review]
Nice work!! Just one suggestion in line, after that ready for AT

File: webapp/src/ts/services/contact-view-model-generator.service.ts (hunk 1 above)
Comment (latin-panda) [reply webapp/src/ts/services/contact-view-model-generator.service.ts:168]: Related, line 154 null check?  `const primaryContact = children.find(child => child.doc?._id === contactId);`
but if all docs have _id 100% then can be ignored :)

Comment (dianabarsan) [reply webapp/src/ts/services/contact-view-model-generator.service.ts:168]: If the doc is undefined, we have biiig problems :D we shouldn't just go on with the execution. 
So this throwing an error should be fine.

**Pull request merged**